x/evm/types/msg.go

```diff
@@ -308,7 +309,38 @@ func (msg MsgEthereumTx) AsTransaction() *ethtypes.Transaction {
 
 // AsMessage creates an Ethereum core.Message from the msg fields
 func (msg MsgEthereumTx) AsMessage(signer ethtypes.Signer, baseFee *big.Int) (core.Message, error) {
-	return msg.AsTransaction().AsMessage(signer, baseFee)
+	txData, err := UnpackTxData(msg.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	gasPrice, gasTipCap, gasFeeCap := txData.GetGasPrice(), txData.GetGasFeeCap(), txData.GetGasTipCap()
+	if baseFee != nil {
+		gasPrice = math.BigMin(gasPrice.Add(gasTipCap, baseFee), gasFeeCap)
+	}
+	var from common.Address
+	if len(msg.From) > 0 {
+		from = common.HexToAddress(msg.From)
+	} else {
+		// heavy path
+		from, err = signer.Sender(msg.AsTransaction())
+		if err != nil {
+			return nil, err
+		}
+	}
+	ethMsg := ethtypes.NewMessage(
+		from,
+		txData.GetTo(),
+		txData.GetNonce(),
+		txData.GetValue(),
+		txData.GetGas(),
+		gasPrice, gasFeeCap, gasTipCap,
+		txData.GetData(),
+		txData.GetAccessList(),
+		false,
+	)
+
+	return &ethMsg, nil
```

Comment:
```suggestion
	return ethMsg, nil
```

Reply:
it returns an interface which is implemented by the pointer.

Reply:
yea, it's the same

Reply:
yes, you are right.